app/api/recommendations.py

```
@@ -1,21 +1,128 @@
+            _log_recommendation_result, 
+            request.user_diagnosis[:100], 
+            len(recommendations), 
+            processing_time,
+            [r.product.id for r in recommendations[:5]]  # 상위 5개 상품 ID만 로깅
+        )
+        
+        return response
+        
+    except Exception as e:
+        logger.error(f"❌ 상품 추천 실패: {e}")
+        raise HTTPException(status_code=500, detail=f"상품 추천에 실패했습니다: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)